### PR TITLE
Binary execution benchmarked. Cereal & Pods.

### PR DESCRIPTION
**Pull request by hidalgoss**
Please, review addings.

### CODE REVIEW

File: benchmark/benchmark_json/include/VectorFloatStruct.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            f_rand_value = get_random_float_using_whole_range();
+            _vec_f[i] = f_rand_value;
+        }
+    }
+
+    size_t getSize(){
+        return _vec_f.size();
+    }
+
+    std::vector<float> get_vec(){
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/include/VectorFloatStruct.h:26]: Name

File: benchmark/benchmark_json/src/BM_Cereal_vs_Pods_Test.cpp [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+        size_t retrievedVectorSize=0;
+
+        VectorFloatStruct4Cereal vec_f(vectorSize);                                         // Load vector randomly with specified size.
+        std::string filename = {JSON_FILE_PATH_AND_NAME + 
+            std::to_string(vectorSize) + "_bin"};                                           // WARNING: Precompiler variable created through root CMakeLists file.
+        vec_f.SerializeVectorToBinaryFile(filename);                                        // Now, use this vector to create a JSON serialization disk file.
+        VectorFloatStruct4Cereal deserializedVectorCereal;                                  // Create an empty vector to load results;
+        for (auto _ : state) 
+        {
+            deserializedVectorCereal.DeserializeBinaryFileToVector(filename);
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Pods_Test.cpp:109]: Same as below. Method names start with lowercase

File: benchmark/benchmark_json/include/SerializationBenchmarkUtils.h [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#ifndef SERIALIZATIONBENCHMARKUTILS_H
+#define SERIALIZATIONBENCHMARKUTILS_H
+
+#include <iostream>
+
+float get_random_float_using_whole_range();
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/include/SerializationBenchmarkUtils.h:6]: Name and make it a member of a class with static methods.

**Review by pablogh-2000 — COMMENTED** [review]
Based on this link: https://github.com/mtrempoltsev/pods#comparison-with-other-libraries
Pods should 4 faster than cereal. Let's review this.

File: benchmark/benchmark_json/include/VectorFloatStruct.h (hunk 1 above)
Comment (hidalgoss) [reply benchmark/benchmark_json/include/VectorFloatStruct.h]: Done.

File: benchmark/benchmark_json/src/BM_Cereal_vs_Pods_Test.cpp [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+        std::string json_string;
+        size_t retrievedVectorSize=0;
+
+        VectorFloatStruct4Cereal vec_f(vectorSize);                                         // Load vector randomly with specified size.
+        std::string filename = {JSON_FILE_PATH_AND_NAME + std::to_string(vectorSize)};      // WARNING: Precompiler variable created through root CMakeLists file.
+        vec_f.SerializeVectorToJsonFile(filename);                                          // Now, use this vector to create a JSON serialization disk file.
+        VectorFloatStruct4Cereal deserializedVectorCereal;                                  // Create an empty vector to load results;
+        for (auto _ : state) 
+        {
+            deserializedVectorCereal.DeserializeJsonFileToVector(filename);
```
[the remark this answers is not in the document]
Comment (hidalgoss) [reply benchmark/benchmark_json/src/BM_Cereal_vs_Pods_Test.cpp]: Done. Question: Constructors & Destructors?? Class names starts in uppercase... Must we rename class names to start lower-cased ?

File: benchmark/benchmark_json/src/BM_Cereal_vs_Pods_Test.cpp (hunk 2 above)
Comment (hidalgoss) [reply benchmark/benchmark_json/src/BM_Cereal_vs_Pods_Test.cpp]: Done.

File: benchmark/benchmark_json/include/SerializationBenchmarkUtils.h (hunk 3 above)
Comment (hidalgoss) [reply benchmark/benchmark_json/include/SerializationBenchmarkUtils.h]: Done.

**Comment by hidalgoss** [comment]
> Based on this link: https://github.com/mtrempoltsev/pods#comparison-with-other-libraries Pods should 4 faster than cereal. Let's review this. We should also test this library: https://github.com/niXman/yas

After lot of proves we can conclude reported marks depends on Data structure used in their cases. When data structure used is out std::vector<float>, marks are not as under their test conditions. We tested their Data class and mark results obtained are as they report but. 

Using YAS, we obtained best marks than Pods reports. 140~168 ns vs 710~760 ns.  22.727G/s throughput vs 5.0218G/s.
YAS must be the choice!

**Pull request closed without merging**